### PR TITLE
Fixes #21241: API to fetch nodes + software times out on large instance

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/21241

This PR improves performance of node API when we fetch softwares, and overall some of the perfs for software related topic
However, it is not sufficient to make improvement enough to get many softwares out of Ruder with the API:
* getting inventory with software from 1 node takes 10s with this change (never ended before).
* getting inventories with software from 111 nodes don't finish

Adding an index on softwareId makes the inventory from 1 node respond in 1 second, and 20s for the 111 nodes

### CODE REVIEW

File: webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/ReadOnlySoftwareInventoryDAOImpl.scala [hunk 1]
```diff
@@ -108,12 +133,25 @@ class ReadOnlySoftwareDAOImpl(
                                  s
                                }
                         } yield {
-                          (NodeId(id), vs)
+                          (NodeId(id), vs.toSeq)
```
Comment (ncharles) [inline webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/ReadOnlySoftwareInventoryDAOImpl.scala:136]: i wanted to use valuesForChunk but it's not there yet in 6.2

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**